docs/developer/developer-handbook.md

```diff
@@ -22,40 +31,299 @@
+UUID, we highly recommend it.
+
+By design, the `Asset` is extensible, so users can store any metadata they want in it. For example, the `properties`
+object could contain a simple string value, or it could be a complex object, following some custom schema. Be aware,
+that unless specified otherwise, all properties are put under the `edc` namespace by default.
+
+Here is an example of how an Asset with a custom property following a custom namespace would look like:
+
+```json
+{
+  "@context": {
+    "edc": "https://w3id.org/edc/v0.0.1/ns/",
+    "ex": "http://w3id.org/example/v0.0.1/ns/"
+  },
+  "@id": "79d9c360-476b-47e8-8925-0ffbeba5aec2",
+  "properties": {
+    "somePublicProp": "a very interesting value",
+    "ex:foo": {
+      "name": "Darth Vader",
+      "webpage": "https://death.star"
+    }
+  }
+  //...
+}
+```
+
+assuming the context object found at http://w3id.org/example/v0.0.1/ns/ contains a type definition of the `foo` object.
+As a reminder: the JSON-LD context could look like this:
+
+```json
+{
+  "@context": {
+    "name": "http://schema.org/name",
+    "webpage": {
+      "@id": "http://schema.org/url",
+      "@type": "@id"
+    }
+  }
+}
+```
+
+Note that upon ingress through the management API, all JSON-LD objects
+get [expanded](https://www.w3.org/TR/json-ld11/#expanded-document-form), because the control plane only operates on
+expanded JSON-LD objects. Since the `Asset` is an extensible type, this is necessary to avoid potential name clashes of
+properties.
+
+This is important to keep in mind, because when querying for assets using the `POST /assets/request` endpoint, the query
+that is submitted in the request body must target the _expanded_ property. For example, a query targeting
+the `somePublicProp` field from the example above would look like this:
+
+```json
+{
+  "https://w3id.org/edc/v0.0.1/ns/filterExpression": [
+    {
+      "https://w3id.org/edc/v0.0.1/ns/operandLeft": [
+        {
+          "@value": "https://w3id.org/edc/v0.0.1/ns/somePublicProp"
+        }
+      ],
+      "https://w3id.org/edc/v0.0.1/ns/operator": [
+        {
+          "@value": "="
+        }
+      ],
+      "https://w3id.org/edc/v0.0.1/ns/operandRight": [
+        {
+          "@value": "a very interesting value"
+        }
+      ]
+    }
+  ]
+}
+```
+
+More information about JSON-LD Contexts and type definitions can be
+found [here](https://www.w3.org/TR/json-ld11/#the-context).
+
+#### Policies
+
+#### ContractDefinitions
+
+#### ContractNegotiations
+
+#### ContractAgreements
+
+#### TransferProcessess
+
+#### A word on JSON-LD contexts
+
 --> explains Assets, Policies, Contract Definitions, etc. from an external API perspective. mentions JSON-LD and related
 specs (ODRL, DCAT)
 
+> The complete OpenAPI specification for the management API is
+> on [SwaggerHub](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.2.0)
```

Comment:
```suggestion
> on [SwaggerHub](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.2.1)
```
it would be great to upload also a "latest" (or similar) version that will get overridden every time, so we don't need to maintain pointers to it

Reply:
maybe we could have just the api name, without the version? i.e. `.../eclipse-edc-bot/management-api`?

Reply:
ok, how about this: lets put the plain `management-api/` link in, with a warning in the text that users must select the latest one. 
Then, we can amend the OpenAPI publishing job in the connector to also create a `latest` version, and once we have that, we can update the docs?